### PR TITLE
[scripts][dependency] Add repository fallback functionality

### PR DESCRIPTION
**Pull request by MahtraDR**
Removing the concept of custom status URLs, and adding fallback functionality so we can rename the master branch, and as a prelude to moving this repo to the Elanthia Online GIthub organization.

### CODE REVIEW

File: dependency.lic [hunk 1]
```diff
@@ -241,25 +241,31 @@ class ScriptManager
...
+  rescue Errno::ENOENT
+    false # false if can't find the server
   end
 
-  def unset_custom_status
-    @status_repo = 'rpherbig/dr-scripts'
-    @status_branch = 'master'
-    update_status_url
+  def update_status_url
+    masterurl = 'https://api.github.com/repos/' + @status_repo + '/git/trees/' + @status_branch
```
Comment (rpherbig) [inline dependency.lic:256]: Given the responses on lines 262 and 264, I'd name these `...main...`  (or `primary`) and `...alternate...`. That one branch is named `master` is a coincidence, I think

File: dependency.lic [hunk 2]
```diff
@@ -241,25 +241,31 @@ class ScriptManager
...
+    # _respond Lich::Messaging.monsterbold("Master URL is #{masterurl.inspect}")
+    mainurl   = 'https://api.github.com/repos/rpherbig/dr-scripts/git/trees/main'
```
Comment (rpherbig) [inline dependency.lic:258]: Shouldn't the fallback URL be `.../master`, which exists right now?

Alternatively, there's almost a fallback to the fallback, `/rpherbig/.../master`, so you could do that now to future-proof this for the duration of the handover.

File: dependency.lic [hunk 3]
```diff
@@ -205,8 +205,8 @@ class ScriptManager
     @paste_bin_token = 'dca351a27a8af501a8d3123e29af7981'
     @paste_bin_url = 'https://pastebin.com/api/api_post.php'
     @firebase_url = 'https://dr-scripts.firebaseio.com/'
-    @status_repo = Settings['status_repo'] || 'rpherbig/dr-scripts'
-    @status_branch = Settings['status_branch'] || 'master'
+    @status_repo = 'rpherbig/dr-scripts'
```
Comment (rpherbig) [inline dependency.lic:208]: If anyone gets stuck on `1.7.4`, you might have them set `Settings['status_branch']` and `Settings['status_repo']` before trying to download a new `dependency.lic`

**Review by rpherbig — APPROVED** [review]

File: dependency.lic (hunk 3 above)
Comment (MahtraDR) [reply dependency.lic:208]: That's a really great idea. Thank you. I always appreciate your thoughtful comments when you review code.

**Comment by MahtraDR** [comment]
Updated the new function to be more descriptive.

image

You now get instant feedback too, when setting a new status url...

image

In theory, we can then change the name of the master branch to main, and or move this repo under EO safely.

**Review by rpherbig — APPROVED** [review]

**Pull request merged**